### DISCUSSION TITLE
Multiple validations on a form element does not work when using material ui

### DISCUSSION THREAD

**Discussion by kamleshwebtech** [react-hook-form/react-hook-form#3779] [Q&A]
I am using material ui for my website ui and also using react-hook-form. My code is as:

```
<TextField
            name="firstName"
            id="firstName"
            label="First Name"
            ref={register({ required: true, minLength: 6 })}
          />
          <div>
            {errors.firstName?.type === "required" && (
              <div className={classes.errorMsg}>First Name is required.</div>
            )}
            {errors.firstName?.type === "minLength" && (
              <div className={classes.errorMsg}>
                First Name length should be more than 6 characters.
              </div>
            )}
          </div>
```

I am trying to apply multiple validations on firstName element but only 'required' is working. I enter any value less than 6 characters then react-hook-form should show 'minLength' error which is not working.

Kindly suggest me this is an issue of react-hook-form or i am doing something wrong to apply. Thanks a lot.

Look forward to hear you.

**Comment by kamleshwebtech** [comment]
I think react-hook-form does not support to Material-Ui's < Textfield /> element. If I use < Input /> tag instead of < Textfield /> tag then multiple validations work while react-hook-form says that it supports Material-UI. Kindly suggest what is the solution of this issue. Thanks.

**Comment by kamleshwebtech** [comment]
@bluebill1049 Please help. Thanks.

**Comment by bluebill1049** [comment]
we do, take a look this section: https://react-hook-form.com/api/#Controller and apply rules

**Reply by kamleshwebtech** [reply]
@bluebill1049  Thanks for your quick reply. I have applied Controller as you suggested but when i submit the form then validation message is displayed in second form submission attempt which should be displayed immediately after first form submission not in second attempt. My code is as:

```
<Grid item container>
          <Controller
            as={TextField}
            name="firstName"
            control={control}
            defaultValue=""
            inputRef={register({ required: true, minLength: 6 })}
          />
          <div>
            {errors.firstName?.type === "required" && (
              <div className={classes.errorMsg}>First Name is required.</div>
            )}
            {errors.firstName?.type === "minLength" && (
              <div className={classes.errorMsg}>
                First Name length should be more than 6 characters.
              </div>
            )}
          </div>
        </Grid>
```

Kindly suggest what is the issue? I am trying to familiar with react-hook-form so that i can use it for my all projects further. 

Look forward to hear you.

**Reply by bluebill1049** [reply]
why you register again? Controller is taking care of register for you.
